plugins/nf-nomad/src/main/nextflow/nomad/NomadConfig.groovy

```diff
@@ -44,19 +44,31 @@ class NomadConfig {
     }
 
     class NomadClientOpts{
+        private Map<String,String> sysEnv
         final String address
         final String token
 
-        NomadClientOpts(Map nomadClientOpts){
-            def tmp = (nomadClientOpts.address?.toString() ?: "http://127.0.0.1:4646")
+        NomadClientOpts(Map nomadClientOpts, Map<String,String> env=null){
+
+            sysEnv = env==null ? new HashMap<String,String>(System.getenv()) : env
+
+            def tmp = (nomadClientOpts.address?.toString() ?: sysEnv.get('NOMAD_ADDR'))
+
             if( !tmp.endsWith("/"))
                 tmp +="/"
             this.address = tmp + API_VERSION
-            token = nomadClientOpts.token ?: null
+            this.token = nomadClientOpts.token ?: sysEnv.get('NOMAD_TOKEN')
+
+            //TODO: Add mTLS properties and env vars
+            // https://developer.hashicorp.com/nomad/docs/commands#mtls-environment-variables
         }
     }
 
     class NomadJobOpts{
+        //TODO: Reevaluate the job level env vars, which are perhaps better suited to be sourced in clientOpts
+        // and overridden for job defs using process directives.
+        private Map<String,String> sysEnv
```

Comment:
same here, use sysEnv in the constructor but we dont need to have a reference to it